go-controller/pkg/allocator/ip/subnet/allocator.go

```diff
@@ -159,6 +161,41 @@ func (allocator *allocator) AllocateUntilFull(name string) error {
 	return nil
 }
 
+// ReserveIPs reserves IPs that are already exist in the cluster. S
+// Should be used for Sync only.
+// We ignore the `IsErrAllocated`, because in case of an error it can reach this function again in reconcile.
+// We break upon any other error, but we don't release IPS.
+// Claims have the IPs, so in case claim is deleted, IP will be released.
+// This way we give another change in the reconcile in case of an error.
```

Comment:
We can take another approach which is all or nothing in the sync, fail in any error, and revert by releasing all what was allocated
since we are under a lock it should be fine
question is what is easier to maintain and simpler